extern/sector-storage/ffiwrapper/sealer_test.go

```diff
@@ -853,3 +854,87 @@ func TestAddPiece512MPadded(t *testing.T) {
 
 	require.Equal(t, "baga6ea4seaqonenxyku4o7hr5xkzbqsceipf6xgli3on54beqbk6k246sbooobq", c.PieceCID.String())
 }
+
+func setupLogger(t *testing.T) *bytes.Buffer {
+	_ = os.Setenv("RUST_LOG", "info")
+
+	var bb bytes.Buffer
+	r, w, err := os.Pipe()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	go func() {
+		_, _ = io.Copy(&bb, r)
+		runtime.KeepAlive(w)
+	}()
+
+	resp := generated.FilInitLogFd(int32(w.Fd()))
+	resp.Deref()
+
+	defer generated.FilDestroyInitLogFdResponse(resp)
+
+	if resp.StatusCode != generated.FCPResponseStatusFCPNoError {
+		t.Fatal(generated.RawString(resp.ErrorMsg).Copy())
+	}
+
+	return &bb
+}
+
+func TestMulticoreSDR(t *testing.T) {
+	if os.Getenv("TEST_RUSTPROOFS_LOGS") != "1" {
+		t.Skip("skipping test without TEST_RUSTPROOFS_LOGS=1")
+	}
+
+	rustLogger := setupLogger(t)
+
+	getGrothParamFileAndVerifyingKeys(sectorSize)
+
+	dir, err := ioutil.TempDir("", "sbtest")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	miner := abi.ActorID(123)
+
+	sp := &basicfs.Provider{
+		Root: dir,
+	}
+	sb, err := New(sp)
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+
+	cleanup := func() {
+		if t.Failed() {
+			fmt.Printf("not removing %s\n", dir)
+			return
+		}
+		if err := os.RemoveAll(dir); err != nil {
+			t.Error(err)
+		}
+	}
+	defer cleanup()
+
+	si := storage.SectorRef{
+		ID:        abi.SectorID{Miner: miner, Number: 1},
+		ProofType: sealProofType,
+	}
+
+	s := seal{ref: si}
+
+	// check multicore
+	_ = os.Setenv("FIL_PROOFS_USE_MULTICORE_SDR", "1")
+	rustLogger.Reset()
+	s.precommit(t, sb, si, func() {})
+
+	ok := false
+	for _, s := range strings.Split(rustLogger.String(), "\n") {
+		if strings.Contains(s, "create_label::multi") {
```

Comment:
can this be used as a hard ✅ ? when the things was down during the weekend, a miner reported that in his log its showing multi-core however, single-core was used in reality

Reply:
Yeah, this tests that the multicore feature works. If it uses only a single core, then that's a separate bug, and I'm not sure if we have a good way to test that from Go